src/test/java/com/redhat/devtools/lsp4ij/fixtures/LSPSemanticTokensFileViewProviderFixtureTestCase.java

```diff
@@ -0,0 +1,289 @@
+    private PsiFile initialize(@NotNull String fileName,
+                               @NotNull String fileBody,
+                               @Nullable String mockSemanticTokensProviderJson,
+                               @Nullable String mockSemanticTokensJson,
+                               boolean enabled) {
+        MockLanguageServer.INSTANCE.setTimeToProceedQueries(100);
+
+        // Must both be non-null or null together
+        assertEquals(mockSemanticTokensProviderJson != null, mockSemanticTokensJson != null);
+
+        ServerCapabilities serverCapabilities = MockLanguageServer.defaultServerCapabilities();
+        SemanticTokensWithRegistrationOptions mockSemanticTokensProvider = mockSemanticTokensProviderJson != null ?
+                JSONUtils.getLsp4jGson().fromJson(mockSemanticTokensProviderJson, SemanticTokensWithRegistrationOptions.class) :
+                null;
+        serverCapabilities.setSemanticTokensProvider(mockSemanticTokensProvider);
+        MockLanguageServer.reset(() -> serverCapabilities);
+
+        SemanticTokens mockSemanticTokens = mockSemanticTokensJson != null ?
+                JSONUtils.getLsp4jGson().fromJson(mockSemanticTokensJson, SemanticTokens.class) :
+                null;
+        MockLanguageServer.INSTANCE.setSemanticTokens(mockSemanticTokens);
+
+        PsiFile file = myFixture.configureByText(fileName, fileBody);
+
+        // Initialize the language server
+        List<LanguageServerItem> languageServers = new LinkedList<>();
+        try {
+            Project project = file.getProject();
+            VirtualFile virtualFile = file.getVirtualFile();
+            ContainerUtil.addAllNotNull(languageServers, LanguageServiceAccessor.getInstance(project)
+                    .getLanguageServers(virtualFile, null, null)
+                    .get(5000, TimeUnit.MILLISECONDS));
+        } catch (Exception e) {
+            fail(e.getMessage());
+        }
+        LanguageServerItem languageServer = ContainerUtil.getFirstItem(languageServers);
+        assertNotNull(languageServer);
+
+        // Enable or disable the view provider as requested
+        LanguageServerDefinition languageServerDefinition = languageServer.getServerDefinition();
+        assertInstanceOf(languageServerDefinition, ClientConfigurableLanguageServerDefinition.class);
+        ClientConfigurableLanguageServerDefinition configurableLanguageServerDefinition = (ClientConfigurableLanguageServerDefinition) languageServerDefinition;
+        ClientConfigurationSettings clientConfiguration = configurableLanguageServerDefinition.getLanguageServerClientConfiguration();
+        assertNotNull(clientConfiguration);
+        clientConfiguration.editor.enableSemanticTokensFileViewProvider = enabled;
+
+        // Force a highlighting pass to populate the view provider's semantic tokens
+        myFixture.doHighlighting();
+
+        return file;
+    }
+
+    protected void assertViewProviderEnabled(@NotNull String fileName,
```

Comment:
And the following two methods are the primary assertion/verification interface for the test subclasses alongside the token verifiers above.